docs/csharp/whats-new/csharp-version-history.md

```diff
@@ -0,0 +1,124 @@
+
+The next version did introduce some interesting new features.
+
+- [Dynamic binding](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/dynamic)
+- [Named/optional arguments](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/named-and-optional-arguments)
+- [Generic covariant and contravariant](https://docs.microsoft.com/en-us/dotnet/standard/generics/covariance-and-contravariance)
+- [Embedded interop types](https://stackoverflow.com/questions/20514240/whats-the-difference-setting-embed-interop-types-true-and-false-in-visual-studi)
+
+Embedded interop types alleviated a deployment pain. Generic covariance and contravariance give you more power to use generics, but they're a bit academic and probably most appreciated by framework and library authors. Named and optional parameters let you eliminate many method overloads and provide convenience. But none of those features are exactly paradigm altering.
+
+The major feature was the introduction of the `dynamic` keyword. The `dynamic` keyword introduced into C# Version 4.0 the ability to override the compiler on compile-time typing. By using the dynamic keyword, you can create constructs similar to dynamically typed languages like JavaScript. You can create a `dynamic x = "a string"` and then add six to it, leaving it up to the runtime to sort out what should happen next.
+
+This gives you the potential for errors but also great power within the language.
+
+## C# version 5.0
+
+C# Version 5.0 was a very focused version of the language. Nearly all of the effort for that version went into another groundbreaking language concept.  Here is the major features list.
+
+- [Asynchronous members](https://msdn.microsoft.com/library/hh191443(vs.110).aspx)
+- [Caller info attributes](https://www.codeproject.com/Tips/606379/Caller-Info-Attributes-in-Csharp)
+
+The caller info attribute lets you easily retrieve information about the context in which you're running without resorting to a ton of boilerplate reflection code. It has many uses in diagnostics and logging tasks.
+
+But `async` and `await` are the real stars of this release. When these features came out in 2012, C# changed the game again by baking asynchrony into the language as a first-class participant. If you've ever dealt with long running operations and the implementation of webs of callbacks, you probably loved this language feature.
+
+## C# version 6.0
+
+With versions 3.0 and 5.0, C# had added some impressive features in an Object Oriented language. (Version 2.0 did as well, but it was fast following Java with those language features.) With version 6.0, it would go away from doing a dominant killer feature and instead release many features that delighted users of the language. Here are some of them.
+
+- [Static imports](http://geekswithblogs.net/BlackRabbitCoder/archive/2015/04/16/c.net-little-wonders-static-using-statements-in-c-6.aspx)
```

Comment:
link to https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-static instead?